blocks/image-placeholder/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { DropZone, FormFileUpload, Placeholder } from '@wordpress/components';
+import { mediaUpload } from '@wordpress/utils';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import MediaUploadButton from '../media-upload-button';
+
+export default function ImagePlaceHolder( { className, setAttributes, icon, label, onSelectImage } ) {
```

Comment:
Great work extracting this, there's something that bother me a bit about the `setAttributes` being passed here and also being passed to `mediaUpload` (probably the root issue).

Instead of passing `setAttributes` to `mediaUpload`, should we refactor it to pass something like `onChange` and `onChange` would be called with a single or multiple images (it's used in gallery aswell).

This would allow us to only pass `onChange` here.

The issue with the current approach is that this "reusable" component enforces the block using it to have certain attribute keys.

Reply:
I totally agree with this changes, the refactor should be applied in the media upload because it is there that we are using the attribute keys. I will add to this PR a refactor to the media upload as previous work.

Reply:
A refactor was applied to the mediaUpload. It is now more generic and does not contain block specific logic e.g: for the gallery.
The placeholder now does not receive setAttributes it just receives onSelectImage that is used for both uploaded and gallery images. An object is passed to function describing the image, the user of the component can pass a function that maps the image object to its attributes.